frontend/src/components/GeneralSetting/AnnouncementSetting.vue

```diff
@@ -0,0 +1,253 @@
+            v-if="!allowEdit"
+            class="text-sm text-gray-400 -translate-y-2 tooltip"
+          >
+            {{ $t("settings.general.workspace.only-owner-can-edit") }}
+          </span>
+        </label>
+        <BBTextField
+          class="mb-5 w-full"
+          :placeholder="'http://xxx.com'"
+          :disabled="!allowEdit"
+          :value="state.announcementSetting.announcementLink"
+          @input="handleExtraDetailsChange"
+        />
+
+        <div class="flex">
+          <button
+            type="button"
+            class="btn-primary ml-auto"
+            :disabled="!allowSave"
+            @click.prevent="updateAnnouncementSetting"
+          >
+            {{ $t("common.update") }}
+          </button>
+        </div>
+      </div>
+    </div>
+
+    <FeatureModal
+      feature="bb.feature.custom-announcement"
+      :open="state.showFeatureModal"
+      @cancel="state.showFeatureModal = false"
+    />
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { cloneDeep, isEqual } from "lodash-es";
+import { computed, reactive, watchEffect } from "vue";
+import { useI18n } from "vue-i18n";
+import { hasFeature, pushNotification, useCurrentUserV1 } from "@/store";
+import { useSettingV1Store } from "@/store/modules/v1/setting";
+import {
+  AnnouncementSetting,
+  AnnouncementSetting_AnnouncementAlertLevel,
+} from "@/types/proto/v1/setting_service";
+import { hasWorkspacePermissionV1 } from "@/utils";
+
+interface LocalState {
+  originalAnnouncementSetting?: AnnouncementSetting;
+  announcementSetting: AnnouncementSetting;
+  showFeatureModal: boolean;
+}
+
+const { t } = useI18n();
+const settingV1Store = useSettingV1Store();
+const currentUserV1 = useCurrentUserV1();
+
+const defaultAnnouncementSetting = function (): AnnouncementSetting {
+  return {
+    announcementLevel:
+      AnnouncementSetting_AnnouncementAlertLevel.ANNOUNCEMENTALERTLEVEL_INFO,
+    announcementText: "",
+    announcementLink: "",
+  };
+};
+
+const state = reactive<LocalState>({
+  announcementSetting: defaultAnnouncementSetting(),
+  originalAnnouncementSetting: undefined,
+  showFeatureModel: false,
+});
+
+watchEffect(() => {
+  const announcementSetting =
+    settingV1Store.workspaceProfileSetting?.announcementSetting;
+  if (announcementSetting) {
+    state.announcementSetting = cloneDeep(
+      settingV1Store.workspaceProfileSetting?.announcementSetting
+    );
+    state.originalAnnouncementSetting = cloneDeep(
+      settingV1Store.workspaceProfileSetting?.announcementSetting
+    );
+  }
+});
+
+const allowEdit = computed((): boolean => {
+  return hasWorkspacePermissionV1(
+    "bb.permission.workspace.manage-general",
+    currentUserV1.value.userRole
+  );
+});
+
+const allowSave = computed((): boolean => {
+  if (!allowEdit.value) {
+    return false;
+  }
+
+  return !isEqual(state.originalAnnouncementSetting, state.announcementSetting);
+});
+
+const handleSelectAlertLevel = (announcementLevel: string) => {
```

Comment:
`<BBSelect>` or `<NSelect>` both support `number` as their value types.
So it is possible to use `AnnouncementSetting_AnnouncementAlertLevel` (number-based enum type) as value type directly to avoid the complex `string-number` type converting.